modules/transitive/src/closure/IdRefVisitor.scala

```diff
@@ -13,23 +13,24 @@
  * limitations under the License.
  */
 
-package closure
+package smithytranslate.closure
 
-import software.amazon.smithy.model.shapes._
+import scala.collection.mutable
+import software.amazon.smithy.model.shapes.*
 import software.amazon.smithy.model.node.Node
 import software.amazon.smithy.model.traits.IdRefTrait
-import scala.jdk.OptionConverters._
-import scala.jdk.CollectionConverters._
+
+import scala.jdk.OptionConverters.*
+import scala.jdk.CollectionConverters.*
 import software.amazon.smithy.model.Model
 import software.amazon.smithy.model.traits.Trait
-import smithytranslate.closure.TransitiveModel
 
-final class IdRefVisitor(
+private[closure] final class IdRefVisitor(
     model: Model,
     value: Node,
     captureTraits: Boolean,
-    validateModel: Boolean,
```

Comment:
removed this argument from here and a few other places. Model validation can be done just at the end, when we have traversed the model

Reply:
absolutely

Reply:
This is most likely binary incompatible, as you're modifying a previously public class

Reply:
I think so too, and we're making private
But at the stage of smithy-translate I don't think we should mind

Reply:
actually, with the repackaging any form of BC goes out of the window anyway 😅